reference/python_types.rst

```diff
@@ -0,0 +1,108 @@
+Mapping NStack types to Python types
+====================================
+
+The table below show you what python types to expect and to return when dealing with types defined in the NStack IDL:
+
+============== ============================ 
+NStack Type    Python Type                
+============== ============================ 
+``Integer``    ``int``              
+``Double``     ``double.Double``  
+``Boolean``    ``bool``  
+``Text``       ``str``   
+Tuple          ``tuple``    
+Struct         ``collections.namedtuple`` *
+Array          ``list``                  
+Sum            see section below         
+``[Byte]``     ``bytes``                  
+``x optional`` ``None`` or x              
+============== ============================
+
+`\*you can return a normal tuple (see Structs section below)`
+
+Sum Types
+---------
+
+Python doesn't have support for Sum types (aka tagged/disjoint unions) but we still support them via inheritance.
+
+Here's some example definitions of type in the nstack IDL::
+
+  type URL = Text;
+  type Referrer = URL;
+  type Target = URL;
+  type Event = PageImpression URL | Click (Referrer, Target);
+
+You can imagine that we create classes for Event and its constructors something like the following::
+
+  class Event(object):
+    ...
+    class PageImpression(Event):
+      ...
+    class Click(Event):
+      ...
+
+As the sum is defined as a type alias (this is the only way to define sums in the IDL) then the types will be available as members of the ``nstack`` package module, and we can construct new values (to return them from an nstack service method) like this:
```

Comment:
"to return them from an nstack service method" in own sentence to make sure we know we're talking about returning, vs. the previous example.